service/controller/internal/nodecontroller/controller.go

```diff
@@ -0,0 +1,106 @@
+package nodecontroller
+
+import (
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/k8sclient/v5/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/v4/pkg/controller"
+	"github.com/giantswarm/operatorkit/v4/pkg/resource"
+	"github.com/giantswarm/operatorkit/v4/pkg/resource/wrapper/retryresource"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/kvm-operator/pkg/label"
+	"github.com/giantswarm/kvm-operator/pkg/project"
+	"github.com/giantswarm/kvm-operator/service/controller/internal/nodecontroller/resource/podcondition"
+	"github.com/giantswarm/kvm-operator/service/controller/key"
+)
+
+type Config struct {
+	Cluster             v1alpha1.KVMConfig
+	ManagementK8sClient client.Client
+	WorkloadK8sClient   k8sclient.Interface
+	Logger              micrologger.Logger
+}
+
+type Controller struct {
+	*controller.Controller
+}
+
+func New(config Config) (*Controller, error) {
+	var err error
+
+	resources, err := newResources(config)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var operatorkitController *controller.Controller
+	{
+		c := controller.Config{
+			K8sClient: config.WorkloadK8sClient,
+			Logger:    config.Logger,
+			Resources: resources,
+			NewRuntimeObjectFunc: func() runtime.Object {
+				return new(corev1.Node)
+			},
+			Selector: labels.SelectorFromSet(map[string]string{
+				label.OperatorVersion: project.Version(),
+				"role":                key.WorkerID,
```

Comment:
Why only worker nodes?

Reply:
We only manage endpoint IPs for workers. We don't remove the master IP even if the node or pod is down because then we wouldn't be able to access the Kubernetes API to check status and add it back once it's healthy again (chicken-and-egg problem).